package.json

```diff
@@ -50,6 +50,7 @@
     "generate-excel-demo-consented": "FR_ENV=demo CCD_DEF_COS_URL=$npm_package_config_demo_cosUrl CCD_DEF_CCD_URL=$npm_package_config_demo_ccdUrl EXCLUDE='*-prod.json' yarn run generate-excel-consented",
     "generate-excel-demo-prod-like-consented": "FR_ENV=demo-prod-like CCD_DEF_COS_URL=$npm_package_config_demo_cosUrl CCD_DEF_CCD_URL=$npm_package_config_demo_ccdUrl EXCLUDE='*-nonprod.json' yarn run generate-excel-consented",
     "generate-excel-aat-consented": "FR_ENV=aat CCD_DEF_COS_URL=$npm_package_config_aat_cosUrl CCD_DEF_CCD_URL=$npm_package_config_aat_ccdUrl EXCLUDE='*-prod.json' yarn run generate-excel-consented",
+    "generate-excel-preview-consented": "FR_ENV=preview CCD_DEF_COS_URL=${process.env.CCD_COS_URL} CCD_DEF_CCD_URL=$npm_package_config_aat_ccdUrl EXCLUDE='*-prod.json' yarn run generate-excel-consented",
```

Comment:
I'd be surprised if that actually worked and `${process.env.CCD_COS_URL}` resolved properly. I think scripts are executed as shell commands so `process.env.` might not resolve properly.

Reply:
I suspect what happens is that you are setting value in `Jenkinsfile_CNP` but then  `CCD_DEF_COS_URL=${process.env.CCD_COS_URL}` overrides it to empty string which makes it not work as you want.

I'd remove `CCD_DEF_COS_URL=${process.env.CCD_COS_URL}` from package.json scripts.